src/benchmark_runner.cc

```diff
@@ -255,7 +253,7 @@ BenchmarkRunner::BenchmarkRunner(
     run_results.file_report_aggregates_only =
         (b.aggregation_report_mode() & internal::ARM_FileReportAggregatesOnly);
     BM_CHECK(FLAGS_benchmark_perf_counters.empty() ||
-             perf_counters_measurement.IsValid())
+             perf_counters_measurement_ptr->IsValid())
```

Comment:
We should add a check that if perf counters are requested, the thread pool has 0 size. I thought we did that check but I don't see it anymore. Anyway - as you pointed out yesterday, it'd be whacky otherwise because each observation would see effects of all threads (unless the counters were set up per thread, which wasn't and isn't the case)

Reply:
I could see the case where someone will look at the combined number of cycles or branch misses on a multi-threaded test, although the utility as you said is questionable. But it would be up to the user to interpret it.  Imho we should not restrict a feature if it does not cause the software to reach an invalid state. Maybe a warning?

Reply:
It is now printing a meaty warning if you select threads and perf counters. 
```
$ test/benchmark_test --benchmark_perf_counters=cycles
2023-03-03T12:45:32-06:00
Running test/benchmark_test
Run on (24 X 4568.16 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x24)
  L1 Instruction 32 KiB (x24)
  L2 Unified 512 KiB (x24)
  L3 Unified 16384 KiB (x8)
Load Average: 4.38, 4.32, 4.24
***WARNING*** CPU scaling is enabled, the benchmark real time measurements may be noisy and will incur extra overhead.
***WARNING*** There are 80 benchmarks with threads and 1 performance counters were requested. Beware counters will reflect the combined usage across all threads.
---------------------------------------------------------------------------------------------------
Benchmark                                         Time             CPU   Iterations UserCounters...
---------------------------------------------------------------------------------------------------
BM_Factorial                                  0.000 ns        0.000 ns   1000000000 cycles=330n 40320
BM_Factorial/real_time                        0.000 ns        0.000 ns   1000000000 cycles=322n 40320
BM_CalculatePiRange/1                          1.32 ns         1.32 ns    531367024 cycles=5.00025 0
BM_CalculatePiRange/8                          10.5 ns         10.5 ns     66417872 cycles=40.002 3.28374
BM_CalculatePiRange/64                         84.3 ns         84.3 ns      8298636 cycles=320.015 3.15746
```